recipe/meta.yaml

```
@@ -21,9 +21,12 @@ build:
   skip: True
   {% endif %}
   script:
-    - cp {{ os.environ["FEEDSTOCK_ROOT"] }}/LICENSE.txt ${RECIPE_DIR}/LICENSE.txt            # [linux]
-    - cp ${RECIPE_DIR}/../LICENSE.txt ${RECIPE_DIR}/LICENSE.txt                              # [osx]
-    - COPY "%RECIPE_DIR%\\..\\LICENSE.txt" "%RECIPE_DIR%\\LICENSE.txt"                       # [win]
+    # Copy feedstock license file to recipe directory
+    - pushd {{ RECIPE_DIR }}
+    - pushd ..
+    - cp LICENSE.txt {{ RECIPE_DIR }}
+    - popd
+    - popd
```

As we mount the recipe directory and the feedstock directory next to each other...

https://github.com/conda-forge/conda-forge-ci-setup-feedstock/blob/252329727dd6cc23116b23cb0f9560c4cf27f095/.scripts/run_docker_build.sh#L87-L88

... `pushd {{ RECIPE_DIR }}/..` would resolve to `/home/conda` (not mention not work on Windows)

To avoid that behavior, we `pushd` to `{{ RECIPE_DIR }}`. Once in `{{ RECIPE_DIR }}`, we are within the feedstock directory's `recipe` directory. So `pushd ..` resolves to the feedstock directory (instead of `/home/conda` as it would have in the other case)

Once we are in the feedstock directory, we can copy the feedstock license file and `popd` back to where we were

This all works around the issue we saw where `FEEDSTOCK_ROOT` didn't resolve during re-rendering

---

Yeah using `.get` works fine too. No preference on which we use but I am very confused as to why this fails now.

---

Agreed it is weird this fails

Though ideally there would be a simple solution so we are not doing different thing for macOS, Linux & Windows

Maybe we can just use `recipe-scripts-license.txt`

```suggestion
    # Copy feedstock license file to recipe directory
    - pushd {{ RECIPE_DIR }}
    - cp recipe-scripts-license.txt LICENSE.txt
    - popd
```